implementation/src/main/java/io/smallrye/metrics/MetricRegistries.java

```diff
@@ -116,21 +120,51 @@ public static MetricRegistry getOrCreate(MetricRegistry.Type type, ApplicationNa
     private static MeterRegistry resolveMeterRegistry(MetricRegistry.Type type) {
         final MeterRegistry meterRegistry;
 
-        if (type == MetricRegistry.Type.BASE) {
-            meterRegistry = new MPPrometheusMeterRegistry(PrometheusConfig.DEFAULT, MetricRegistry.Type.BASE);
-            meterRegistry.config().commonTags("scope", MetricRegistry.Type.BASE.getName());
-            meterRegistry.config().meterFilter(mpMeterBaseRegistryAccessFilter);
-
-        } else if (type == MetricRegistry.Type.APPLICATION) {
-            meterRegistry = new MPPrometheusMeterRegistry(PrometheusConfig.DEFAULT, MetricRegistry.Type.APPLICATION);
-            meterRegistry.config().commonTags("scope", MetricRegistry.Type.APPLICATION.getName());
-            meterRegistry.config().meterFilter(mpMeterAppRegistryAccessFilter);
-        } else if (type == MetricRegistry.Type.VENDOR) {
-            meterRegistry = new MPPrometheusMeterRegistry(PrometheusConfig.DEFAULT, MetricRegistry.Type.VENDOR);
-            meterRegistry.config().commonTags("scope", MetricRegistry.Type.VENDOR.getName());
-            meterRegistry.config().meterFilter(mpMeterVendorRegistryAccessFilter);
+        if (isCDIAvailable()) {
+            final BeanManager bm = CDI.current().getBeanManager();
+            final List<MeterRegistry> meterRegistries = new ArrayList<>();
+
+            final Set<Bean<?>> beans = bm.getBeans(MeterRegistry.class, MicrometerBackends.class.getAnnotation(Backend.class));
+            for (Bean<?> bean : beans) {
+                final Object reference = bm.getReference(bean, MeterRegistry.class, bm.createCreationalContext(bean));
+                if (MeterRegistry.class.isInstance(reference)) {
+                    meterRegistries.add(MeterRegistry.class.cast(reference));
```

Comment:
Is it possible to keep the MPPrometheusMeterRegistry as is and just add the "new" registries to  the Global Registry directly?

Seems like all the registries are being included to the BASE, APPLICATION, VENDOR `MPCompositeMeterRegistry` and those  3 composite registries  will then be added to the global meter registry anyways. Specifically, wouldn't this mean the same single Prometheus Registry has all app, base and vendor (and future custom scope) metrics. This would be difficult to filter out with the existing scrape calls that the Prometheus client provides for the /metric?scope=___ calls.

The CounterAdapter, GaugeAdapter, etc are already registering with the Global Meter Registry directly and there are the MeterFilters made for each Prometheus Meter Registry scope (that we control) available that helps with registering the right metric for the right scope (but will register to anything else in the global registry).


Also, as part of the custom scope change, I was intending to change this to : https://github.com/Channyboy/smallrye-metrics/blob/micrometer-customScopes/implementation/src/main/java/io/smallrye/metrics/MetricRegistries.java#L111

Reply:
@Channyboy Thanks for the review! That sounds reasonable, let me take a swing at adding these to the global registry.

Reply:
I've made the change - however - what I now see is that the application metrics from the various annotated methods do not show up in my other configured backend systems, whereas they did when using the MPCompositeRegistry previously. Having those metrics show up in Elastic / Statsd / whatever is definitely what I was shooting for. Any thoughts?

Reply:
@jgallimore  Sorry for the delay, these should help https://github.com/smallrye/smallrye-metrics/pull/511